cmake/linker/iar/config_file_script.cmake

```
@@ -0,0 +1,922 @@
+      # Something to init
+      create_symbol(OBJECT ${REGION_OBJECT} SYMBOL __${name}_load_start
+        EXPR "@ADDR(${first_section_name}_init)@"
+        )
+    else()
+      create_symbol(OBJECT ${REGION_OBJECT} SYMBOL __${name}_load_start
+        EXPR "@LOADADDR(${first_section_name})@"
+        )
+    endif()
+
+    create_symbol(OBJECT ${REGION_OBJECT} SYMBOL __${name}_start
+      EXPR "@ADDR(${first_section_name})@"
+      )
+    create_symbol(OBJECT ${REGION_OBJECT} SYMBOL __${name}_end
+      EXPR "@END(${last_section_name})@"
+      )
+    create_symbol(OBJECT ${REGION_OBJECT} SYMBOL __${name}_size
+      EXPR "(@(__${name}_end)@ - @(__${name}_start)@)"
+      )
+
+  endforeach()
+
+  get_property(symbols GLOBAL PROPERTY ${REGION_OBJECT}_SYMBOLS)
+  foreach(symbol ${symbols})
+    get_property(name GLOBAL PROPERTY ${symbol}_NAME)
+    get_property(expr GLOBAL PROPERTY ${symbol}_EXPR)
+    if(NOT DEFINED expr)
+      create_symbol(OBJECT ${REGION_OBJECT} SYMBOL __${name}_size
+        EXPR "@(ADDR(${name})@"
+        )
+    endif()
+  endforeach()
+
+  # This is only a trick to get the memories
+  set(groups)
+  get_objects(LIST groups OBJECT ${REGION_OBJECT} TYPE GROUP)
+  foreach(group ${groups})
+    get_property(group_type  GLOBAL PROPERTY ${group}_OBJ_TYPE)
+    get_property(parent      GLOBAL PROPERTY ${group}_PARENT)
+    get_property(parent_type GLOBAL PROPERTY ${parent}_OBJ_TYPE)
+
+    if(${group_type} STREQUAL GROUP)
+      get_property(group_name GLOBAL PROPERTY ${group}_NAME)
+      get_property(group_lma  GLOBAL PROPERTY ${group}_LMA)
+      if(${group_name} STREQUAL ROM_REGION)
+        set_property(GLOBAL PROPERTY ILINK_ROM_REGION_NAME ${group_lma})
+      endif()
+    endif()
+
+    if(${parent_type} STREQUAL GROUP)
+      get_property(vma GLOBAL PROPERTY ${parent}_VMA)
+      get_property(lma GLOBAL PROPERTY ${parent}_LMA)
+
+      set_property(GLOBAL PROPERTY ${group}_VMA ${vma})
+      set_property(GLOBAL PROPERTY ${group}_LMA ${lma})
+    endif()
+  endforeach()
+
+endfunction()
+
+#
+# String functions - start
+#
+
+function(system_to_string)
+  cmake_parse_arguments(STRING "" "OBJECT;STRING" "" ${ARGN})
+
+  get_property(name    GLOBAL PROPERTY ${STRING_OBJECT}_NAME)
+  get_property(regions GLOBAL PROPERTY ${STRING_OBJECT}_REGIONS)
+  get_property(format  GLOBAL PROPERTY ${STRING_OBJECT}_FORMAT)
+
+  # Ilink specials
+  # set(${STRING_STRING} "build for rom;\n")
+  set(${STRING_STRING} "build for ram;\n")
+  if("${format}" MATCHES "aarch64")
+    set(${STRING_STRING} "${${STRING_STRING}}define memory mem with size = 16E;\n")
+  else()
+    set(${STRING_STRING} "${${STRING_STRING}}define memory mem with size = 4G;\n")
+  endif()
+
+  foreach(region ${regions})
+    get_property(name    GLOBAL PROPERTY ${region}_NAME)
+    get_property(address GLOBAL PROPERTY ${region}_ADDRESS)
+    get_property(flags   GLOBAL PROPERTY ${region}_FLAGS)
+    get_property(size    GLOBAL PROPERTY ${region}_SIZE)
```

```suggestion
    get_property(name GLOBAL PROPERTY ${region}_NAME)
    get_property(address GLOBAL PROPERTY ${region}_ADDRESS)
    get_property(flags GLOBAL PROPERTY ${region}_FLAGS)
    get_property(size GLOBAL PROPERTY ${region}_SIZE)
```
also needs applying throughout